roachtest: sqlsmith/setup=seed/setting=no-mutations failed

roachtest.sqlsmith/setup=seed/setting=no-mutations [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6166285?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6166285?buildTab=artifacts#/sqlsmith/setup=seed/setting=no-mutations) on master @ [cb55144cdec54d2a70f074ad64b4eca5e6c6891a](https://github.com/cockroachdb/cockroach/commits/cb55144cdec54d2a70f074ad64b4eca5e6c6891a):


```
test artifacts and logs in: /artifacts/sqlsmith/setup=seed/setting=no-mutations/run_1
	sqlsmith.go:303,sqlsmith.go:325,test_runner.go:896: ping node 1: dial tcp 35.243.224.110:26257: connect: connection refused
		HINT: node likely crashed, check logs in artifacts > logs/1.unredacted
		previous sql:
		SELECT
			crdb_internal.replication_stream_spec(crdb_internal.create_sql_schema_telemetry_job()::INT8::INT8)::BYTES
				AS col_73510
		LIMIT
			1:::INT8;ping node 1: dial tcp 35.243.224.110:26257: connect: connection refused
		HINT: node likely crashed, check logs in artifacts > logs/1.unredacted
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=seed/setting=no-mutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-18785

roachtest.sqlsmith/setup=seed/setting=no-mutations [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6195961?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6195961?buildTab=artifacts#/sqlsmith/setup=seed/setting=no-mutations) on master @ [80c274877a917580af62be6eb0cd48c8c7ae9c08](https://github.com/cockroachdb/cockroach/commits/80c274877a917580af62be6eb0cd48c8c7ae9c08):


```
test artifacts and logs in: /artifacts/sqlsmith/setup=seed/setting=no-mutations/run_1
	sqlsmith.go:265,sqlsmith.go:325,test_runner.go:896: error: pq: internal error: unable to vectorize execution plan: overlaps(localtimestamp("$0"), "$3", "$3", '2031-06-13 18:44:18.000563+00:00':::TIMESTAMPTZ): unknown signature: overlaps(timestamp, timestamptz, timestamptz, timestamptz)
		stmt:
		SELECT
			count(*) AS col_253061
		FROM
			defaultdb.public.seed@[0] AS tab_102827
		WHERE
			overlaps(localtimestamp(tab_102827._int8::INT8):::TIMESTAMPTZ::TIMESTAMPTZ::TIMESTAMPTZ, tab_102827._timestamptz::TIMESTAMPTZ, tab_102827._timestamptz::TIMESTAMPTZ, '2031-06-13 18:44:18.000563+00:00':::TIMESTAMPTZ::TIMESTAMPTZ)::BOOL
		GROUP BY
			tab_102827._int8, tab_102827._date, tab_102827._float8
		ORDER BY
			tab_102827._float8, tab_102827._int8 DESC, tab_102827._date ASC, tab_102827._date ASC
		LIMIT
			7:::INT8;
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=seed/setting=no-mutations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/logictestccl/testdata/logic_test/crdb_internal
+++ b/pkg/ccl/logictestccl/testdata/logic_test/crdb_internal
@@ -176,3 +176,6 @@ SELECT crdb_internal.stream_ingestion_stats_json(unique_rowid());
 
 query error pq: crdb_internal\.stream_ingestion_stats_json\(\): job.*is not a stream ingestion job
 SELECT crdb_internal.stream_ingestion_stats_json(id) FROM (SELECT id FROM system.jobs LIMIT 1);
+
+query error pq: crdb_internal\.replication_stream_spec\(\): job.*is not a replication stream job
+SELECT crdb_internal.replication_stream_spec(crdb_internal.create_sql_schema_telemetry_job())

--- a/pkg/ccl/streamingccl/streamproducer/stream_lifetime.go
+++ b/pkg/ccl/streamingccl/streamproducer/stream_lifetime.go
@@ -213,7 +213,11 @@ func getReplicationStreamSpec(
 	planCtx := dsp.NewPlanningCtx(evalCtx.Ctx(), jobExecCtx.ExtendedEvalContext(),
 		nil /* planner */, noTxn, sql.DistributionTypeSystemTenantOnly)
 
-	replicatedSpans := j.Details().(jobspb.StreamReplicationDetails).Spans
+	details, ok := j.Details().(jobspb.StreamReplicationDetails)
+	if !ok {
+		return nil, errors.Errorf("job with id %d is not a replication stream job", streamID)
+	}
+	replicatedSpans := details.Spans
 	spans := make([]roachpb.Span, 0, len(replicatedSpans))
 	for _, span := range replicatedSpans {
 		spans = append(spans, *span)

